src/avatar/avatar.component.ts

```diff
@@ -23,49 +21,42 @@ export enum Size {
     selector: "igx-avatar",
     templateUrl: "avatar.component.html"
 })
-export class IgxAvatarComponent implements AfterViewInit, AfterContentChecked {
+export class IgxAvatarComponent implements OnInit {
     @ViewChild("image") public image: ElementRef;
     @ViewChild("initialsImage") public initialsImage: ElementRef;
-    @Input() public initials: string;
-    @Input() public src: string;
-    @Input("roundShape") public roundShape = "false";
-    @Input() public color = "white";
+    @ViewChild("imageTemplate", { read: TemplateRef }) protected imageTemplate: TemplateRef<any>;
+    @ViewChild("initialsTemplate", { read: TemplateRef }) protected initialsTemplate: TemplateRef<any>;
+    @ViewChild("iconTemplate", { read: TemplateRef }) protected iconTemplate: TemplateRef<any>;
 
     @HostBinding("attr.aria-label") public ariaLabel = "avatar";
     @HostBinding("attr.role") public role = "img";
     @HostBinding("attr.class")
     public get classes() {
-        return "igx-avatar igx-avatar--" + this.size;
+        if (this.isRounded) {
+            return `igx-avatar--rounded igx-avatar--${this.size}`;
+        }
+        return `igx-avatar igx-avatar--${this.size}`;
     }
 
     public sizeEnum = Size;
     public roleDescription: string;
-
-    @ViewChild("imageTemplate", { read: TemplateRef }) protected imageTemplate: TemplateRef<any>;
-    @ViewChild("initialsTemplate", { read: TemplateRef }) protected initialsTemplate: TemplateRef<any>;
-    @ViewChild("iconTemplate", { read: TemplateRef }) protected iconTemplate: TemplateRef<any>;
-
     private _size: string;
     private _bgColor: string;
-    private _icon = "android";
-    private _isInitialsRepositionNeeded = false;
+    private _icon: string;
+    private _initials: string;
+    private _src: string;
+
+    @Input("roundShape") public roundShape = "false";
+    @Input() public color = "";
 
     @Input()
     get size(): string {
-        if (this._isInitialsRepositionNeeded) {
-            this.repositionInitials();
-        }
-
         return this._size === undefined ? "small" : this._size;
     }
 
     set size(value: string) {
         const sizeType = this.sizeEnum[value.toUpperCase()];
```

Comment:
Typescript supports string enumerations. Initialize the `Size` enumeration with the strings for the size and this check is kind of obsolete.